packages/admin-next/dashboard/src/routes/orders/returns-create/components/create-returns/returns-form.tsx

```diff
@@ -0,0 +1,410 @@
+                  </Form.Hint>
+                  <Form.ErrorMessage />
+                </Form.Item>
+              )
+            }}
+          />
+        </div>
+
+        <div className="mt-10 flex flex-col gap-y-4">
+          <Form.Field
+            control={form.control}
+            name="enable_custom_refund"
+            render={({ field }) => {
+              return (
+                <Form.Item>
+                  <div className="flex items-center justify-between">
+                    <Form.Label>{t("orders.returns.customRefund")}</Form.Label>
+                    <Form.Control>
+                      <Form.Control>
+                        <Switch
+                          checked={!!field.value}
+                          onCheckedChange={field.onChange}
+                        />
+                      </Form.Control>
+                    </Form.Control>
+                  </div>
+                  <Form.Hint className="!mt-1">
+                    {t("orders.returns.customRefundHint")}
+                  </Form.Hint>
+                  <Form.ErrorMessage />
+                </Form.Item>
+              )
+            }}
+          />
+
+          {form.watch("enable_custom_refund") && (
+            <div className="w-[50%] pr-2">
+              <Form.Field
+                control={form.control}
+                name="custom_refund"
+                render={({ field: { onChange, ...field } }) => {
+                  return (
+                    <Form.Item>
+                      <Form.Control>
+                        <CurrencyInput
+                          min={0}
+                          max={order.refundable_amount}
+                          onValueChange={onChange}
+                          code={order.currency_code}
+                          symbol={getCurrencySymbol(order.currency_code)}
+                          {...field}
+                        />
+                      </Form.Control>
+                      <Form.ErrorMessage />
+                    </Form.Item>
+                  )
+                }}
+              />
+            </div>
+          )}
+        </div>
+
+        <div className="mt-10 flex flex-col gap-y-4">
+          <Form.Field
+            control={form.control}
+            name="enable_custom_shipping_price"
+            render={({ field }) => {
+              return (
+                <Form.Item>
+                  <div className="flex items-center justify-between">
+                    <Form.Label
+                      tooltip={t("orders.returns.shippingPriceTooltip")}
+                    >
+                      {t("orders.returns.customShippingPrice")}
+                    </Form.Label>
+                    <Form.Control>
+                      <Form.Control>
+                        <Switch
+                          disabled={
+                            form.watch("enable_custom_refund") ||
+                            !form.watch("shipping")
+                          }
```

Comment:
suggestion: Since this is also disabled if the user selects a custom refund amount we need to update the tooltip accordingly